src/Standards/BestIt/Sniffs/Functions/FluentSetterSniff.php

```diff
@@ -118,56 +130,81 @@ protected function processTokenWithinScope(
             return;
         }
 
-        $nextReturnPtr = $phpcsFile->findNext(
-            T_RETURN,
-            $returnPtr + 1,
-            $closeBracePtr
+        $this->checkAndRegisterEmptyReturnErrors($phpcsFile, $functionPos, $returnPtr, $errorData);
+    }
+
+    /**
+     * Get the sniff name.
+     *
+     * @param string $sniffName If there is an optional sniff name.
+     *
+     * @return string Returns the special sniff name in the code sniffer context.
+     */
+    private function getSniffName(string $sniffName = ''): string
+    {
+        $sniffFQCN = preg_replace(
+            '/Sniff$/',
+            '',
+            str_replace(['\\', '.Sniffs'], ['.', ''], static::class)
         );
 
-        if ($nextReturnPtr !== false) {
-            $phpcsFile->addError(
-                self::ERROR_MULTIPLE_RETURN_FOUND,
-                $stackPtr,
-                self::CODE_MULTIPLE_RETURN_FOUND,
-                $errorData
-            );
-            return;
+        if ($sniffName) {
+            $sniffFQCN .= '.' . $sniffName;
         }
 
-        $thisVariablePtr = $phpcsFile->findNext(T_VARIABLE, $returnPtr, null, false, '$this', true);
-
-        if ($thisVariablePtr === false) {
-            $fixMustReturnThis = $phpcsFile->addFixableError(
-                self::ERROR_MUST_RETURN_THIS,
-                $stackPtr,
-                self::CODE_MUST_RETURN_THIS,
-                $errorData
-            );
+        return $sniffFQCN;
+    }
 
-            if ($fixMustReturnThis) {
-                $this->fixMustReturnThis($phpcsFile, $returnPtr);
-            }
+    /**
+     * Processes the tokens that this test is listening for.
+     *
+     * @param File $file The file where this token was found.
+     * @param int $functionPos The position in the stack where this token was found.
+     * @param int $classPos The position in the tokens array that opened the scope that this test is listening for.
+     *
+     * @return void
+     */
+    protected function processTokenWithinScope(
+        File $file,
+        $functionPos,
+        $classPos
+    ): void {
+        $isSuppressed = SuppressHelper::isSniffSuppressed(
```

Comment:
Hauptsache in der Original-Methode n early return und der Rest der Methode war "unschön" ...